mem/mem.py

```diff
@@ -0,0 +1,80 @@
+from common.configurable_model import ConfigurableModel
+import functools
+from bit_vector import BitVector
+from enum import Enum
+
+
+class Mode(Enum):
+    LINE_BUFFER = 0
+    FIFO = 1
+    SRAM = 2
+
+
+def check_addr(fn):
```

Comment:
I see what you're saying now. There's no way to move this into the class? or at least the `gen_mem` closure?